api/candlepin-api-spec.yaml

```diff
@@ -31,6 +31,34 @@ paths:
         default:
           $ref: '#/components/responses/default'
 
+  /openapi.json:
```

Comment:
I am not sure if we need to serve these dynamically. Once you delete our current static Swagger UI under server/src/main/webapp and instead deploy the latest swagger version as described in the "Plain old HTML/CSS/JS (Standalone)" section here: https://swagger.io/docs/open-source-tools/swagger-ui/usage/installation/ then that should be able to serve both the UI and the json file alone. That should be enough (not sure it's worth it adding an extra endpoint just for the yaml)

Reply:
I thought our goal was to deprecate ```/candlepin/swagger.json``` end point, which is used to generate the API spec by scanning the code therefore I created these endpoints to serve openAPI spec directly. 
In recent changes 
- I removed the ```/candlepin/openapi.json``` endpoint since the generated spec have issues with (offsetdatetime, default response).
- Kept the ```/candlepin/openapi.yaml``` endpoint since we need fetch raw OpenAPI spec (Just like /swagger.json endpoint).
- Updated to new Swagger UI to render ```candlepin-api-spec.yaml```.

Reply:
Yes, when I wrote this comment I wasn't yet aware that swagger.json has issues.

Although I still don't think we need the `/candlepin/openapi.yaml` endpoint, since the yaml is still served statically, and is downloadable from the /docs page. You can either click on it on the Swagger UI page, or just do `curl -k https://localhost:8443/candlepin/docs/candlepin-api-spec.yaml` to get it.

There is still the question though of serving the swagger.json. Personally, I don't think we should serve it since it has the problems you mentioned, but maybe it still has value (because some people might prefer to parse json instead of yaml)? (I think this might be a topic of discussion with the team).